definitions/npm/ramda_v0.x.x/flow_v0.82.x-/test_ramda_v0.x.x_logic.js

```diff
@@ -61,11 +72,58 @@ const def1: number = defaultTo42(undefined);
 const feither = _.either(gt10, even);
 const feitherR: boolean = f(101);
 
-const incCount = _.ifElse(_.is(Number), _.inc, _.toString);
-const ie: number | string = incCount({});
-const ie2: number | string = incCount(1);
-
-const em: boolean = _.isEmpty([1, 2, 3]);
+describe('ifElse', () => {
+  it('uses a union of both branches as the return type', () => {
+    const incCount = ifElse(x => x % 2 == 0, x => x + 1, x => x.toString());
+    const ie: number | string = incCount(1);
+  })
+
+  it('checks the condition input matches the branch inputs', () => {
+    const incCount = ifElse(x => x % 2 == 0, x => x + 1, x => x.toString());
+    // Because the object literal ({}) isn't a number it
+    // $ExpectError
+    const ie: number | string = incCount({});
+  })
+
+  it('works with variadic arg inputs', () => {
+    const ifElseFn = ifElse(
+      // The original motivation was to work with "test". Future changes could
+      // invalidate the nature of the test, so we need to mimic the form of test
+      // at time of writing. The error will be along the lines of this:
```

Comment:
I'm a little confused because we're writing a test about `test` in the context of `ifElse`. Would you mind expanding a bit here and disambiguating? I'm still having trouble understanding this point.

Reply:
Ah - this is perhaps the wrong kind of context I'm providing. The original ticket this is from is essentially "`test` does not work with `ifElse`", when really the problem is `ifElse` doesn't work with varargs. This might have worked as a self-made PR comment instead of a comment in the code, since the ticket itself and a PR comment are ephemeral to each other compared to the code that will outlast the others' relevances.

I'll remove this comment.

Reply:
Okay this should be addressed now. There is a comment there still, but the `test` context is now gone.